workflow/controller/pod/controller.go

```diff
@@ -113,7 +113,7 @@ func (c *Controller) HasSynced() func() bool {
 func (c *Controller) Run(ctx context.Context, workers int) {
 	defer c.workqueue.ShutDown()
 	go c.podInformer.Run(ctx.Done())
```

Comment:
The podInformer.Run() call will start the pod informer, which can still make incorrect decisions based on the unsynced wfInformer. I think this Run() needs to move after the waitforcachesync on wfInfomer.

Reply:
If we move `podInformer.Run()` below the `waitforcachesync`, will `podInformer.HasSynced` return `true`?

Reply:
I think we'll have to call WaitForCacheSync twice, once for `wfInformer`, then `podInformer.Run`, then wait on the podInformer in a second call.

Reply:
resolved in 6b0aaf39d465037da697ab9a1cf35a8067a28fba

Reply:
Thanks